examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java

```diff
@@ -0,0 +1,287 @@
+    String uniqueServerName = "in-process server for " + getClass();
+    features = new ArrayList<Feature>();
+    server = new RouteGuideServer(InProcessServerBuilder.forName(uniqueServerName), 0, features);
+    server.start();
+    inProcessChannel = InProcessChannelBuilder.forName(uniqueServerName).build();
+  }
+
+  @After
+  public void tearDown() throws InterruptedException {
+    inProcessChannel.shutdownNow();
+    server.stop();
+  }
+
+  @Test
+  public void testGetFeature() {
+    Point point = Point.newBuilder().setLongitude(1).setLatitude(1).build();
+    Feature unnamedFeature = Feature.newBuilder()
+        .setName("").setLocation(point).build();
+    RouteGuideGrpc.RouteGuideBlockingStub stub = RouteGuideGrpc.newBlockingStub(inProcessChannel);
+
+    // feature not found in the server
+    Feature feature = stub.getFeature(point);
+
+    assertEquals(unnamedFeature, feature);
+
+    // feature found in the server
+    Feature namedFeature = Feature.newBuilder()
+        .setName("name").setLocation(point).build();
+    features.add(namedFeature);
+
+    feature = stub.getFeature(point);
+
+    assertEquals(namedFeature, feature);
+  }
+
+  @Test
+  public void testListFeatures() throws InterruptedException {
+    // setup
+    Point lo = Point.newBuilder().setLongitude(0).setLatitude(0).build();
+    Point hi = Point.newBuilder().setLongitude(10).setLatitude(10).build();
+    Rectangle rect = Rectangle.newBuilder().setLo(lo).setHi(hi).build();
+    Point p1 = Point.newBuilder().setLongitude(-1).setLatitude(-1).build();
+    Point p2 = Point.newBuilder().setLongitude(2).setLatitude(2).build();
+    Point p3 = Point.newBuilder().setLongitude(3).setLatitude(3).build();
+    Point p4 = Point.newBuilder().setLongitude(4).setLatitude(4).build();
+    Feature f1 = Feature.newBuilder().setLocation(p1).setName("f1").build(); // not inside rect
+    Feature f2 = Feature.newBuilder().setLocation(p2).setName("f2").build();
+    Feature f3 = Feature.newBuilder().setLocation(p3).setName("f3").build();
+    Feature f4 = Feature.newBuilder().setLocation(p4).build(); // unamed
+    features.add(f1);
+    features.add(f2);
+    features.add(f3);
+    features.add(f4);
+    final Collection<Feature> result = new HashSet<Feature>();
+    final CountDownLatch latch = new CountDownLatch(1);
+    StreamObserver<Feature> responseObserver =
+        new StreamObserver<Feature>() {
+          @Override
+          public void onNext(Feature value) {
+            result.add(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            fail();
+          }
+
+          @Override
+          public void onCompleted() {
+            latch.countDown();
+          }
+        };
+    RouteGuideGrpc.RouteGuideStub stub = RouteGuideGrpc.newStub(inProcessChannel);
+
+    // run
+    stub.listFeatures(rect, responseObserver);
+    latch.await();
+
+    // verify
+    Collection<Feature> expected = new HashSet<Feature>();
```

Comment:
FYI: I find it convenient to use `Arrays.asList`: `new HashSet<Feature>(Arrays.asList(f2, f3));`. Doesn't work as well when the element type has a generic, but this case works fine.